Sources/SpatialiteObjC/LibSpatialite.m

```diff
@@ -0,0 +1,12 @@
+#import <Foundation/Foundation.h>
+#import "include/LibSpatialite.h"
+
+void setupSpatialite(sqlite3 *db_handle){
+  // spatialite_init(0);
+  puts("SpatiaLite integrated");
+
+  const void *ptr;
+  int verbose;
+
+  spatialite_init_ex (db_handle, ptr, verbose);
```

Comment:
check out: https://www.gaia-gis.it/gaia-sins/splite-doxy-5.0.1/demo1_8c-example.html

looks like we need to call it as:
```
    cache = spatialite_alloc_connection ();
    spatialite_init_ex (handle, cache, 0);
```

Reply:
i'd recommend some like guarding in this file. let me know what you think:

```
// 1. declare the cache point at the top of the file (outside the function)
void *cache = 0;

// 2. in the setupSpatialite function, check if the cache is created and bail if it is
if cache != 0 {
  puts("Already initialized, bailing);
  return;
}
// ... proceed to call the init function

// 3. add a cleanup function
void cleanupSpatialite() {
  if cache != 0 {
    spatialite_cleanup_ex(cache);
    cache = 0;
  }
}

Reply:
we can hold off on this for now - this is mainly helpful for event store unit tests, and i'm not exactly sure what's needed to make that work properly here. we can cross that bridge as we integrate